File: files/CH-088.py
import pandas as pd

path = "CH-088 Subtotal Calculation.xlsx"
input = pd.read_excel(path, usecols = "B:E", skiprows = 1, nrows = 16)
test = pd.read_excel(path, usecols="I:M", skiprows=1, nrows=21)
test.columns = test.columns.str.replace(".1", "")
pivot_table = (
    input.melt(id_vars=["Product", "Season"], var_name="Region", value_name="value")
    .pivot_table(values="value", index=["Product", "Season"], columns="Region", aggfunc="sum")
    .reset_index()
)

subtotals = pivot_table.groupby("Product").sum(numeric_only=True).reset_index()
grand_total = pivot_table.sum(numeric_only=True).to_frame().T
pivot_table = pd.concat([pivot_table, subtotals, grand_total], ignore_index=True)
pivot_table["Total Regions"] = pivot_table.loc[:, pivot_table.columns[2:]].sum(axis=1)
pivot_table = pivot_table.sort_values(by=["Product", "Season"], ignore_index=True)

print(pivot_table.columns[2:].all() == test.columns[2:].all()) # True
# only numerical values compared, didn't check labels.

File: files/000-099/088/CH-088.R
library(tidyverse)
library(readxl)
library(reshape2)

path = "files/CH-088 Subtotal Calculation.xlsx"
input = read_excel(path, range = "B2:E18")
test  = read_excel(path, range = "I2:M23")

result = input %>%
  pivot_longer(cols = -c(1, 2), names_to = "Region", values_to = "value") %>%
  dcast(Product + Season ~ Region, 
                     value.var = "value", 
                     fun.aggregate = sum, 
                     margins = c("Product", "Season")) %>%
  mutate(`Total Regions` = rowSums(.[, -c(1, 2)], na.rm = TRUE)) %>%
  as_tibble()

all.equal(result[,3:5], test[,3:5], check.attributes = FALSE)
# [1] TRUE - result the same, didn't change total labels
